zad2/warriors.rb

```diff
@@ -1,12 +1,75 @@
-class Warrior
-  def initialize(name:, level:)
+class Character
+  attr_accessor :name, :level
+  def initialize(name: , level:)
+    @name = name
+    @level = valid_level(level)
   end
 
   def strength
+    @level + [*1..12].sample
+  end
+  
+  #we make sure that level is in range 1..99
+  def valid_level(lvl)
+    lvl.between?(1,99) ? lvl : (lvl > 99 ? 99 : 1)
+  end
+
+  def card
+    puts("#{@name}, lvl #{@level}")
+  end
+
+end
+
+class Warrior < Character
+  def points(p)
+    #level increases only if the opponent's level was lower
+    if p > 0 
+      @level = valid_level(@level + p)
+    end
   end
 end
 
+class Monster < Character
+end
+
 class BattleArena
   def initialize(first_character, second_character)
+    @first_character = first_character
+    @second_character = second_character
+    puts("A fight between #{@first_character.name} and #{@second_character.name}")
+  end
+
+  def battle!
+    puts("#{@first_character.name} attacked #{@second_character.name} with #{att1 = @first_character.strength} damage.")
+    puts("#{@second_character.name} attacked #{@first_character.name} with #{att2 = @second_character.strength} damage.")
+    att1 == att2 ? puts("Draw.") : (att1 > att2 ? winner(@first_character, @second_character) : winner(@second_character, @first_character))
```

Comment:
`condition ? do_a : do_b` używamy tylko w przypadku bardzo prostych wyrażeń.
W tym przypadku to trochę ukrywanie logiki bo niby jest jedna linijka, a tak naprawdę to podwójny zagnieżdżony `if/else` 😱

Reply:
a jak widać, że logiki w metodzie jest za dużo to wiadomo co można poprawić 👍